### PR TITLE
RGW - Zipper - Remove a number of casts from rgw_admin

### PR DESCRIPTION
**Pull request by dang**
There are still a ton of casts to RadosStore in rgw_admin.  Remove the
easy ones.  Many of the rest represent actual operations that are
specific to RadosStore, and need to be split out.

Signed-off-by: Daniel Gryniewicz <dang@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_sal_rados.cc [hunk 1]
```diff
@@ -3019,6 +3019,11 @@ int RadosZoneGroup::get_zone_by_name(const std::string& name, std::unique_ptr<Zo
   return 0;
 }
 
+int RadosZoneGroup::list_zones(const DoutPrefixProvider* dpp, std::list<std::string>& zone_ids)
+{
+  return store->svc()->zone->list_zones(dpp, zone_ids);
```
Comment (cbodley) [inline src/rgw/rgw_sal_rados.cc:3024]: this lists all zones in the cluster, right? i'd expect a `ZoneGroup::list_zones()` to only list the zones in that zonegroup. maybe we need a better name or place for this?

**Review by cbodley — APPROVED** [review]

**Review by soumyakoduri — APPROVED** [review]

**Comment by dang** [comment]
jenkins test make check

**Comment by dang** [comment]
jenkins test api

**Comment by dang** [comment]
https://pulpito.ceph.com/dang-2022-08-31_12:26:08-rgw-wip-dang-zipper-admin-distro-default-smithi/

Only known failures

**Pull request merged**